src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -546,6 +604,9 @@ private void OnBeforeClosing()
 
         private void SolutionSaveAs_AfterExecute(string Guid, int ID, object CustomIn, object CustomOut)
         {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            UpdateSolutionDirectory();
```

Comment:
In `IntializeAsync` method we are only invoking `UpdateSolutionDirectory` method if solution is open. In this case, we are invoking it directly. I hope that is okay but just wanted to check.

https://github.com/NuGet/NuGet.Client/pull/4879/files#diff-bed53a8b2e737e49ff6141c4279e525dfa2d100eabf6510ef6e0a1746de10505R217-R220

Reply:
I don't understand what "we are invoking it directly" means, or rather why it might be of concern.

Reply:
Oh, I understand the question now.  In this case, I'll give my answer in the form of a question:  How can a solution be saved if no solution if open? 😁

Reply:
> How can a solution be saved if no solution if open? 😁

Good question that answers my question.